### DISCUSSION TITLE
use sqitch in existing deployments

### DISCUSSION THREAD

**Discussion by jmls** [theory/sqitch#835] [General]
So, I've been manually using pgMigrate and have come across sqithc. Loving the idea of proper version control :) 

However, as this is an existing app, how do I init sqitch with the existing schema ? I do not want )(obviously) to have to create manual deploy and revert scripts for every table, function etc

To add a slight twist , there are a couple of deployments that are not in sync with the latest dev version - how would I bring these into sync with development if I switch to sqitch ?

**Comment by theory** [comment]
Hi @jmls. There's actually [a FAQ for that](https://github.com/sqitchers/sqitch/wiki/How-do-I-get-started-with-an-existing-database%3F)!

For your second question, you might consider starting with a dump of your production database for the first change, and then create a second with just the current changes in dev. How you diff those depends on your current tooling, though. Hopefully it's not too many changes.

I have, in the past, put in the effort of turning a database dump into individual changes. Usually took a couple days, but it all depends on the complexity. IIRC, a while back @adunstan started work on a script to parse a schema-only [`pg_dump`](https://www.postgresql.org/docs/current/app-pgdump.html) file and turn it into changes, but I don't know if it got beyond the experimental phase.

HTH,

David

**Comment by adunstan** [comment]
On 2024-09-18 We 2:29 PM, David E. Wheeler wrote:
>
> Hi @jmls <https://github.com/jmls>. There's actually a FAQ for that 
> <https://github.com/sqitchers/sqitch/wiki/How-do-I-get-started-with-an-existing-database%3F>!
>
> For your second question, you might consider starting with a dump of 
> your production database for the first change, and then create a 
> second with just the current changes in dev. How you diff those 
> depends on your current tooling, though. Hopefully it's not too many 
> changes.
>
> I have, in the past, put in the effort of turning a database dump into 
> individual changes. Usually took a couple days, but it all depends on 
> the complexity. IIRC, a while back @adunstan 
> <https://github.com/adunstan> started work on a script to parse a 
> schema-only |pg_dump| 
> <https://www.postgresql.org/docs/current/app-pgdump.html> file and 
> turn it into changes, but I don't know if it got beyond the 
> experimental phase.
>

Umm, nope, don't think so.

Message ID: 
***@***.***>

**Comment by theory** [comment]
On Sep 18, 2024, at 14:50, Andrew Dunstan ***@***.***> wrote:

> Umm, nope, don't think so.

Really? I could have sworn. Would have been around 10 years ago though.
